core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -1036,27 +1038,41 @@ wasm_interp_call_func_import(WASMModuleInstance *module_inst,
 #if WASM_ENABLE_DEBUG_INTERP != 0
 #define CHECK_SUSPEND_FLAGS()                                          \
     do {                                                               \
+        WASMCluster *cluster = wasm_exec_env_get_cluster(exec_env);    \
+        if (cluster)                                                   \
+            os_mutex_lock(&cluster->lock);                             \
```

Comment:
`cluster->lock` is used to lock the whole exec_env list in the cluster, using it here is a little heavy and may cause dead lock (e.g., other thread is suspending all other threads in future GC feature, which may lock `cluster->lock` firstly).

Suggest to use `exec_env->wait_lock` instead. And had better add `exec_env->wait_lock` for the related read/write operations of suspend_flag/signal_flag?